src/components/ObsEdit/AddID.js

```diff
@@ -0,0 +1,193 @@
+// @flow
+
+import * as React from "react";
+import {
+  View,
+  Pressable,
+  TouchableOpacity, FlatList, Image
+} from "react-native";
+import {useNavigation} from "@react-navigation/native";
+import { viewStyles, textStyles } from "../../styles/obsDetails/addID";
+import { useTranslation } from "react-i18next";
+import { TextInput as NativeTextInput } from "react-native";
+import AddIDHeader from "./AddIDHeader";
+import {useRef, useState} from "react";
+import {
+  BottomSheetBackdrop,
+  BottomSheetModal,
+  BottomSheetModalProvider
+} from "@gorhom/bottom-sheet";
+import {Button, Headline, Text, TextInput} from "react-native-paper";
+import Icon from "react-native-vector-icons/MaterialCommunityIcons";
+import useRemoteSearchResults from "../../sharedHooks/useRemoteSearchResults";
+import ViewNoFooter from "../SharedComponents/ViewNoFooter";
+import {colors} from "../../styles/global";
+import uuid from "react-native-uuid";
+
+type Props = {
+  route: {
+    params: {
+      onIDAdded: ( identification: {[string]: any} ) => void,
+      goBackOnSave: boolean
+    }
+  }
+}
+
+
+const AddID = ( { route }: Props ): React.Node => {
+  const { t } = useTranslation( );
+  const [comment, setComment] = useState( "" );
+  const [commentDraft, setCommentDraft] = useState( "" );
+  const { onIDAdded, goBackOnSave } = route.params;
+  const bottomSheetModalRef = useRef( null );
+  const [taxonSearch, setTaxonSearch] = useState( "" );
+  const taxonList = useRemoteSearchResults( taxonSearch, "taxa", "taxon.name,taxon.preferred_common_name,taxon.default_photo.square_url,taxon.rank" ).map( r => r.taxon );
```

Comment:
`.map( r => r.taxon )` doesn't seem to be necessary here and is causing a crash later on whenever you search for... anything. So

1. Remove it
2. Add an integration test for `AddID` where `inatjs.search` is mocked but `useRemoteSearchResults` is not to ensure this component is always correctly handling the results from `useRemoteSearchResults`

Reply:
Regarding the Promise rejection - this is happening regardless of AddID (for example, same issue is happening on `main` branch). Should I look into this?
From looking into this, it seems related to an issue with `saveLocalObservationForUpload` while calling `objectForPrimaryKey`.
This means the Realm instance has been closed, but weirdly enough - it's been open right before this, so not sure why it's failing here

Reply:
Ok, if the promise rejection isn't new to this branch, let's deal with it in a separate issue.